$NetBSD$

Avoid sudo -i to make the scripts much more usable.
Make sure the Erlang cookie file is locked down.
--- rel/files/riak.orig	2012-03-07 22:08:10.000000000 +0000
+++ rel/files/riak
@@ -23,8 +23,7 @@ if [ "$RUNNER_USER" -a "x$LOGNAME" != "x
         echo "sudo doesn't appear to be installed and your EUID isn't $RUNNER_USER" 1>&2
         exit 1
     fi
-    echo "Attempting to restart script through sudo -u $RUNNER_USER"
-    exec sudo -u $RUNNER_USER -i $RUNNER_SCRIPT_DIR/$RUNNER_SCRIPT $@
+    exec sudo -u $RUNNER_USER $RUNNER_SCRIPT_DIR/$RUNNER_SCRIPT $@
 fi
 
 # Warn the user if ulimit -n is less than 1024
@@ -98,6 +97,12 @@ case "$1" in
             echo "Node is already running!"
             exit 1
         fi
+
+        # Cookie file should exist now, set proper permissions
+        if [ -f $PLATFORM_DATA_DIR/.erlang.cookie ]; then
+            chmod 400 $PLATFORM_DATA_DIR/.erlang.cookie
+        fi
+
         # Sanity check the app.config file
         RES=`$NODETOOL_LITE chkconfig $RUNNER_ETC_DIR/app.config`
         if [ "$RES" != "ok" ]; then
